lib/evmone/eof.cpp

```diff
@@ -349,40 +349,41 @@ bool validate_rjump_destinations(bytes_view code) noexcept
 std::variant<EOFValidationError, int32_t> validate_max_stack_height(
     bytes_view code, size_t func_index, const std::vector<EOFCodeType>& code_types)
 {
-    assert(!code.empty());
-
     // Special values used for detecting errors.
     static constexpr int32_t LOC_UNVISITED = -1;  // Unvisited byte.
-    static constexpr int32_t LOC_IMMEDIATE = -2;  // Immediate byte.
 
     // Stack height in the header is limited to uint16_t,
     // but keeping larger size for ease of calculation.
-    std::vector<int32_t> stack_heights(code.size(), LOC_UNVISITED);
-    stack_heights[0] = code_types[func_index].inputs;
+    struct StackHeightRange
+    {
+        int32_t min = LOC_UNVISITED;
+        int32_t max = LOC_UNVISITED;
```

Comment:
```suggestion
        int32_t max = LOC_UNVISITED;
        
        bool visited() const {...}
```

This should simplify some `LOC_VISITED` handling later.

Reply:
Now you can hide the constant in the `StackHeightRange::UNVISITED` :)

Reply:
Look like static members not allowed in local structs.
```
/home/andrei/dev/evmone/lib/evmone/eof.cpp:357:30: error: static data member 'LOC_UNVISITED' not allowed in local struct 'StackHeightRange'
        static const int32_t LOC_UNVISITED = -1;
                             ^
1 error generated.
```